test/unit/app/app.test.js

```diff
@@ -626,6 +630,97 @@ describe('App',  () => {
 
       it('should return the current patient in view as patient and empty permissions', () => {
         expect(result.patient).to.deep.equal(Object.assign({}, loggedIn.allUsersMap.d4e5f6, { permissions: {} }));
+      }); 
+
+      const clinicianupload = {
+        allUsersMap: {
+          a1b2c3: {
+            termsAccepted: 'today'
+          },
+          d4e5f6: {}
+        },
+        currentPatientInViewId: 'd4e5f6',
+        loggedIn: true,
+        loggedInUserId: 'a1b2c3',
+        notification: {
+          key: 'fetchingPatient',
+          link: {
+            to: '/patients/foo',
+            text: 'Sorry!'
+          },
+          status: 405
+        },
+        permissionsOfMembersInTargetCareTeam: {},
+        membershipPermissionsInOtherCareTeams: {
+          d4e5f6: {
+            view: {},
+            note: {},
+            upload: {},
+          }
+        },
+        dataDonationAccounts: [],
+        datasources: [],
+        showingDonateBanner: null,
+        showingDexcomConnectBanner: null,
+        working: {
+          fetchingUser: {inProgress: false},
+          fetchingPendingSentInvites: {inProgress: false},
+          updatingDataDonationAccounts: {inProgress: false},
+          fetchingPatient: {inProgress: false, notification: {type: 'error'}},
+          loggingOut: {inProgress: false}
+        }
+      };
+      const clinicianuploadresult = mapStateToProps({blip: clinicianupload});
+
+      it('should return user is clinician with upload permissions', () => {
```

Comment:
Since we'll have the context already provided in the containing block, I suggest simplifying the `it` statements as much as possible for all the tests.  i.e:
```javascript
it('should map the correct `permsOfLoggedInUser` permissions', () => {});
```